packages/js/src/generators/library/library.ts

```diff
@@ -262,17 +264,13 @@ function createFiles(tree: Tree, options: NormalizedSchema, filesDir: string) {
     tmpl: '',
     offsetFromRoot: offsetFromRoot(options.projectRoot),
     rootTsConfigPath: getRelativePathToRootTsConfig(tree, options.projectRoot),
-    buildable: options.buildable === true,
+    buildable: options.bundler && options.bundler !== 'none',
     hasUnitTestRunner: options.unitTestRunner !== 'none',
   });
 
-  if (options.compiler === 'swc') {
+  if (options.bundler === 'swc' || options.bundler === 'rollup') {
```

Comment:
question: what if I want to bundle with `rollup` but still keep the default `babel` of `rollup:rollup` executor?

Reply:
To be honest, I tried to keep the behaviour unchanged. In the past, if you set `bundler` to `rollup`, it would default to `swc`. Since we want to deprecate the `rollup + tsc` I thought that this would be the only option. But I can remove the `|| bundler === 'rollup'` if we think it's wrong here!

Reply:
Ok, after our chat I got this:

Now, the current behaviour is that when a user chooses `rollup` as bundler, the `compiler` option in their `project.json` build target is left blank, which would default to `babel` in the `rollup` executor.

In this new way of doing things, we are forcing the `compiler` option in their build target to be `swc`. So we want to allow the option to leave that blank. Is this correct?

Reply:
@jaysoo wdyt?

Reply:
Babel is only used by React libs and we want to encourage users to use:

- Node/plain JS libs -> tsc/swc/esbuild
- React/Web -> vite/rollup

And for rollup, we should default to SWC here since it's much faster.

Reply:
Ok, so no changes needed here